### PR TITLE
Change widget previews

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This PR aligns widget previews with their updated designs. Apart from regular preview bitmaps I added layout previews that are supported since SDK 31. This greatly improves preview experience as it enables couple of things:

1. Previews scale to the available space.
2. Previews use dynamic colors.
3. They can use localized texts for previews.

In 1b3e2f708d8d0460fd4822504943ed2217008992 I updated secondary texts alpha in the actual widgets to `0.7` to align it with the designs.

## Testing Instructions

Verify widget previews on devices pre and post SDK 31.

## Screenshots or Screencast 

> [!note]
> To generate these screenshots I temporarily removed our legacy widget so that all new widgets fit in a single screenshot.

### Previews

| Bitmap light | Bitmap dark | XML light | XML dark |
| - | - | - | - |
| Screenshot_20240508_205634 | Screenshot_20240508_205619 | Screenshot_20240508_204445 | Screenshot_20240508_204505 |

### Fixed sizing issue

| Before | After |
| - | - |
| bitmap | layout | 

## Checklist

- [x] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] ~Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)~
- [x] ~I have considered whether it makes sense to add tests for my changes~
- [x] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [x] ~Any jetpack compose components I added or changed are covered by compose previews~
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] ~for accessibility with TalkBack~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/localization/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -165,6 +165,12 @@
     <string name="widget_no_episode_playing">No episode playing</string>
     <string name="widget_nothing_in_up_next">Nothing in Up Next</string>
     <string name="widget_check_out_discover">Check out Discover for more</string>
+    <string name="widget_preview_episode_title">Episode title</string>
+    <string name="widget_preview_episode_in_up_next_title">Episode in Up Next</string>
+    <string name="widget_preview_time_left_1">25m left</string>
+    <string name="widget_preview_time_left_2">12m left</string>
+    <string name="widget_preview_time_left_3">55m left</string>
+    <string name="widget_preview_time_left_4">1h 36m left</string>
```
Comment (mebarbosa) [inline modules/services/localization/src/main/res/values/strings.xml:173]: I think we can remove `widget_preview_time_left_4` since we are not using

**Review by mebarbosa — APPROVED** [review]
👋🏽 Nice work! I disabled auto merge because I left one small comment. I am not sure if you will handle on this PR or not. But I am already approving it

File: modules/features/widgets/src/main/res/layout/widget_player_large_preview.xml [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+                android:textStyle="bold" />
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="4dp"
+                android:alpha="0.7"
+                android:ellipsize="end"
+                android:maxLines="1"
+                android:text="@string/widget_preview_time_left_3"
```
Comment (MiSikora) [inline modules/features/widgets/src/main/res/layout/widget_player_large_preview.xml:388]:
```suggestion
                android:text="@string/widget_preview_time_left_4"
```

File: modules/services/localization/src/main/res/values/strings.xml (hunk 1 above)
Comment (MiSikora) [reply modules/services/localization/src/main/res/values/strings.xml:173]: Good catch! It is used now f501e12fe7bf4f41ca8ecc8134d9f3fa20f4b7c2. I accidentally used `widget_preview_time_left_3` twice.

**Pull request merged**